File: CPP/Program/ParkingLot.cpp
// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>
#include "Vehicle.cpp"

// Deklarasi namespace
using namespace std;


/*
    Kelas ParkingLot yang digunakan untuk merepresentasikan data ParkingLot.
    Parking lot merupakan composite untuk Vehicle
*/
class ParkingLot {
private:
    // Atribut yang digunakan pada kelas ParkingLot terdiri dari 
    //kapasitas, jumlah kendaraan saat ini dan daftar kendaraan 
    int kapasitas;
    int jumlahKendaraanSaatIni;
    vector<Vehicle*> daftarKendaraan;
public:
    // Constructor kosong
    ParkingLot()
    {


    }

    // Constructor :yg diisi parameter
    ParkingLot(int kapasitas, int jumlahKendaraanSaatIni)
    {
        this->kapasitas = kapasitas;
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
    }


    //membuat getter setter
    void setkapasitas(int kapasitas)
    {
        this->kapasitas = kapasitas;
    }


    void setjumlahKendaraanSaatIni(int jumlahKendaraanSaatIni)
    {
        this->jumlahKendaraanSaatIni = jumlahKendaraanSaatIni;
    }

    int getkapasitas()
    {
        return kapasitas;
    }


    int getjumlahKendaraanSaatIni()
    {
        return jumlahKendaraanSaatIni;
    }

    //membuat fungsi untuk cek jumlah kendaraan dengan kapasitas kendaraan di garasi
    void addVehicle(Vehicle* kendaraan) {
        if (jumlahKendaraanSaatIni < kapasitas) {
            daftarKendaraan.push_back(kendaraan);
            jumlahKendaraanSaatIni++;
        } else {
            cout << "Parking lot penuh" << endl;
        }
    }

    //mmembuat fungsi untuk menampilkan jumlah kendaraan terparkir dan kapasitas
    void displayStatus() const {
        cout << "Jumlah Kendaraan terparkir: " << jumlahKendaraanSaatIni << " Kendaraan" << endl;
        cout << "Kapasitas: " << kapasitas << endl;
    }

    // Destructor
    ~ParkingLot()
    {


    }
};

File: Python/Program/ParkingLot.py
from Vehicle import Vehicle

class ParkingLot:
    def __init__(self, kapasitas=0, jumlah_kendaraan_saat_ini=0):  #inisialisasi atribut
        self.kapasitas = kapasitas  #kapasitas tempat parkir
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini  #jumlah kendaraan saat ini
        self.daftar_kendaraan = []  #daftar kendaraan yang terparkir

    def set_kapasitas(self, kapasitas):  #metode untuk mengatur kapasitas tempat parkir
        self.kapasitas = kapasitas

    def set_jumlah_kendaraan_saat_ini(self, jumlah_kendaraan_saat_ini):  #metode untuk mengatur jumlah kendaraan saat ini
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini

    def get_kapasitas(self):  #metode untuk mendapatkan kapasitas tempat parkir
        return self.kapasitas

    def get_jumlah_kendaraan_saat_ini(self):  #metode untuk mendapatkan jumlah kendaraan saat ini
        return self.jumlah_kendaraan_saat_ini

    def add_vehicle(self, kendaraan):  #metode untuk menambahkan kendaraan ke tempat parkir
        if self.jumlah_kendaraan_saat_ini < self.kapasitas:  #jika masih ada tempat parkir kosong
            self.daftar_kendaraan.append(kendaraan)  #tambahkan kendaraan ke daftar
            self.jumlah_kendaraan_saat_ini += 1  #tambahkan 1 ke jumlah kendaraan saat ini
        else:
            print("Parking lot penuh")  #jika tempat parkir penuh, cetak pesan

    def display_status(self):  #metode untuk menampilkan status tempat parkir
        print("Jumlah Kendaraan terparkir:", self.jumlah_kendaraan_saat_ini, "Kendaraan")  #tampilkan jumlah kendaraan terparkir
        print("Kapasitas:", self.kapasitas)  #tampilkan kapasitas tempat parkir